### PR TITLE
Increase test coverage

### PR DESCRIPTION
**Pull request by daandouwe**
Let's increase the test coverage.

There are easy fixes and there are hard fixes.

### Easy
Easy places to put more tests:
- [x] `kiwi/training/optimizers.py` (54/143 lines missing — mostly `DenseSparseAdam` which **is not used**)
- [x] `kiwi/modules/common/scorer.py` (34/72 lines missing — mostly `DotProductScorer`, `GeneralScorer` and `OperationScorer` which are **not used anywhere** in Kiwi)
- [x] `kiwi/modules/common/layer_norm.py` (15/23 lines missing — `LayerNorm` is **not used anywhere** (`TFLayerNorm` is, by the PredEst model)
- [ ] `kiwi/modules/common/positional_encoding.py` (15/21 lines missing — all in `PositionalEncoding` which is used in the PredEst model)
- [x] `kiwi/modules/sentence_level_output.py` (18 lines missing, all in `SentenceFromLogits`, `SentenceScoreRegression`, and `BinarySentenceScore`)
- [x] `kiwi/utils/io.py` (35/84 lines missing)
- [x] `kiwi/cli.py` (20/34 missing)
- [x] `kiwi/loggers.py` (54/85 lines missing)
- [x] `kiwi/utils/tensors.py` (16 lines missing — bunch of random functions that are (**not used anywhere**)

### Hard(er)
Bunch of harder ones:
- [x] `kiwi/utils/migrations/v0_to_v2.py` (42 lines missing, mostly in `convert_state_dict_from_v0_to_v2_for_predictor`)
- [x] `kiwi/metrics/metrics.py` (98 (!) lines missing — things like `ThresholdCalibrationMetric`, `MovingMetric`, `MovingF1`, `MovingSkipsAtQuality`.)
- [x] `kiwi/data/batch.py` (29/49 lines missing)
- [ ] `normalize_prediction_files` in `kiwi/lib/evaluate.py` (~20 lines and important!)
- [ ] `predict_from_configuration` and `setup_run` in `kiwi/lib/evaluate.py` (~25 lines and important!)
- [ ] `kiwi/data/datasets/parallel_dataset.py`, `kiwi/data/datasets/wmt_qe_dataset.py`, `kiwi/data/encoders/parallel_data_encoder.py`, and `kiwi/data/encoders/wmt_qe_data_encoder.py`
- [x] `kiwi/systems/qe_system.py` (118 missing!)

### CODE REVIEW

**Comment by daandouwe** [comment]
Master coverage: 
[htmlcov-master.zip](https://github.com/Unbabel/OpenKiwi/files/5402595/htmlcov-master.zip)

New coverage:
[Uploading htmlcov.zip…]()

File: kiwi/modules/common/scorer.py [hunk 1]
```diff
@@ -175,48 +94,48 @@ def forward(self, query, keys):
...
+#     out = OperationScorer(query_size, query_size, attn_size, op='add')(q, q)
+#     assert list(out.shape) == [batch_size, q.shape[1], q.shape[1]]
+#
+#     out = MLPScorer(
+#         query_size, keys_size, layer_sizes=[10, 5, 5], activation=nn.Sigmoid
+#     )(q, ks)
+#     assert list(out.shape) == [batch_size, q.shape[1], ks.shape[1]]
+#
+#     out = MLPScorer(query_size, keys_size, layer_sizes=[10, 5, 5])(q, ks)
+#     assert list(out.shape) == [batch_size, q.shape[1], ks.shape[1]]
```
Comment (kepler) [inline kiwi/modules/common/scorer.py:141]: Just remote it. :)
```suggestion

```

File: tox.ini [hunk 2]
```diff
@@ -25,7 +25,7 @@ skip_install = true
 usedevelop = true
 envdir = {toxworkdir}/py37
 commands =
-    poetry install -v
+    poetry install -v -E mlflow
```
Comment (kepler) [inline tox.ini:28]: Why do you mlflow here?

**Review by kepler — CHANGES_REQUESTED** [review]

File: tox.ini (hunk 2 above)
Comment (daandouwe) [reply tox.ini:28]: I added a test in which train uses mlflow. Hits all the `MLflowLogger` code :)

**Review by kepler — APPROVED** [review]
